File: Python/8kyu/8kyu-grasshopper-personalized-message.py
# Grasshopper - Personalized Message
# Language: Python
# Version: 3.10
# Difficulty: 8 kyu

def greet(name, owner):
    return "Hello boss" if name == owner else "Hello guest"


# Tests
import codewars_test as test
from solution import greet

@test.describe("Fixed Tests")
def fixed_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(greet('Daniel', 'Daniel'), 'Hello boss')
        test.assert_equals(greet('Greg', 'Daniel'), 'Hello guest')

File: R/8kyu/8kyu-grasshopper-personalized-message.r
# Grasshopper - Personalized Message
# Language: R
# Version: 3.4.1
# Difficulty: 8 kyu

# Solution using ifelse()
greet <- function(name, owner){
  ifelse(name == owner, "Hello boss", "Hello guest")
}

# Solution using if()
greet <- function(name, owner){
  if (name == owner){
    "Hello boss"
    }else{
    "Hello guest"
  }
}


# Tests
test_that('Basic tests', {
  expect_equal(greet('Daniel', 'Daniel'), 'Hello boss')
  expect_equal(greet('Greg', 'Daniel'), 'Hello guest')
})